File: tests/machine/x/pascal/exists_builtin.pas
program ExistsBuiltin;
{$mode objfpc}
{$modeswitch nestedprocvars}

uses SysUtils, fgl, fphttpclient, Classes, Variants, fpjson, jsonparser;

type
  generic TArray<T> = array of T;

var
  _tmp0: specialize TArray<integer>;
  data: specialize TArray<integer>;
  flag: boolean;
  x: integer;

begin
  data := specialize TArray<integer>([1, 2]);
  SetLength(_tmp0, 0);
  for x in data do
    begin
      if not ((x = 1)) then continue;
      _tmp0 := Concat(_tmp0, [x]);
    end;
  flag := (Length(_tmp0) > 0);
  writeln(flag);
end.

File: tests/machine/x/cpp/exists_builtin.cpp
#include <algorithm>
#include <iostream>
#include <vector>

int main() {
  std::vector<int> data = std::vector<int>{1, 2};
  auto flag =
      std::any_of(data.begin(), data.end(), [&](auto x) { return (x == 1); });
  std::cout << std::boolalpha << flag << std::endl;
  return 0;
}
